src/rtos/rtos.c

```diff
@@ -468,10 +468,7 @@ int rtos_get_gdb_reg(struct connection *connection, int reg_num)
 {
 	struct target *target = get_target_from_connection(connection);
 	int64_t current_threadid = target->rtos->current_threadid;
-	if ((target->rtos != NULL) && (current_threadid != -1) &&
```

Comment:
Just curious why the current_threadid checks are no longer needed in the "if" statement

Reply:
That was something I did that turned out not to be necessary. I've reverted this change.